### PR TITLE
Added BlockFormListener, disallowed pvp protection players to enter end, added chat filter, disabled tnt carts / primed TNT going in portals

### PR DESCRIPTION
**Pull request by GangstaDev**
Prevents Cobble monstering

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -13,4 +13,14 @@ nerf-endermen-drops: false
 nerf-pigzombie-drops: false
 animal-chunk-cap: 30
 endpoint: http://standardsurvival.com/api/v1/
-rts-address: http://rts.standardsurvival.com
\ No newline at end of file
+rts-address: http://rts.standardsurvival.com
+
+muted-words:
+- 'nigger'
```
Comment (sbezboro) [inline config.yml]: leave this list empty, I don't want these words committed in the repo

File: src/com/sbezboro/standardplugin/listeners/PlayerChatListener.java [hunk 2]
```diff
@@ -14,99 +14,115 @@
...
+            for (int i = 0; i < helpTopics.length; i++) {
+                if (helpTopics[i].matches(message)) {
+                    helpTopics[i].sendHelpMessage(player);
+                    return;
+                }
+            }
+        }
+        boolean sendWarning = false;
+        char star = ("*").charAt(0);
+        for (String str : plugin.getConfig().getStringList("muted-words")) {
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/listeners/PlayerChatListener.java]: I use a custom config system in both plugins rather than accessing the Config directly (see the `StandardConfig`) class, all config settings should be defined there, then exposed through `StandardPlugin`

File: src/com/sbezboro/standardplugin/listeners/PlayerChatListener.java [hunk 3]
```diff
@@ -14,99 +14,115 @@
...
+        if (player.getTimeSpent() < 600) {
+            for (int i = 0; i < helpTopics.length; i++) {
+                if (helpTopics[i].matches(message)) {
+                    helpTopics[i].sendHelpMessage(player);
+                    return;
+                }
+            }
+        }
+        boolean sendWarning = false;
+        char star = ("*").charAt(0);
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/listeners/PlayerChatListener.java]:
```suggestion
        char star = '*';
```

File: src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java [hunk 4]
```diff
@@ -52,8 +52,9 @@ public void onPlayerPortal(PlayerPortalEvent event) {
 				if (newEnd != null) {
 					event.setTo(new Location(newEnd, 100, 50, 0));
 				}
-				
-				plugin.getLogger().info(player.getDisplayName(false) + " going to the end.");
+				if (!(player.isPvpProtected())) {
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java]: The line below is just logging to console, no need to restrict

File: src/com/sbezboro/standardplugin/listeners/PlayerTeleportListener.java [hunk 5]
```diff
@@ -11,49 +13,55 @@
...
+            }
 
+            if (blockedPearlMaterials.contains(to.getBlock().getType())) {
+                event.setCancelled(true);
+            }
+        }
+        if (player.isPvpProtected() && event.getCause() == END_PORTAL) {
+            event.setCancelled(true);
+            player.sendMessage(ChatColor.DARK_PURPLE + "You cannot enter the end while PvP Protected! Use " +
+                    ChatColor.AQUA + "/enablepvp" + ChatColor.DARK_PURPLE + " to remove your PvP Protection!");
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/listeners/PlayerTeleportListener.java]: why this change?

**Review by sbezboro — COMMENTED** [review]
Thanks for the PR! Some comments

File: src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java (hunk 4 above)
Comment (GangstaDev) [reply src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java]: Added this so it doesn't log to console when PvP protected players try to enter end, the idea behind blocking pvp prot players from entering end is alt abusing during end resets with pvp timer

File: config.yml (hunk 1 above)
Comment (GangstaDev) [reply config.yml]: Sorry about that, should've thought about that

File: src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java (hunk 4 above)
Comment (sbezboro) [reply src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java]: Why remove the console log? Only I ever see it, just gives me an idea of the flow player in and out of the end

Comment (Tincon1) [reply src/com/sbezboro/standardplugin/listeners/PlayerPortalListener.java]: oh, I am stupid.... I did if they are not pvp protected my bad, will change

File: src/com/sbezboro/standardplugin/listeners/PlayerTeleportListener.java (hunk 5 above)
Comment (GangstaDev) [reply src/com/sbezboro/standardplugin/listeners/PlayerTeleportListener.java]: Didn’t mean to click resolve convo on your yesterday’s comment, but I added if the player is not pvp protected to only log then, due to the adding of pvp protected players not being able to enter end so alts can’t abuse protection in end, I can remove both of those if you want

**Comment by GangstaDev** [comment]
The logger showing the flow in and out of end still works, just tested

**Comment by sbezboro** [comment]
Thanks so much!
